src/nlp/features.py

```diff
@@ -458,7 +460,10 @@ def generate_from_arrow_type(pa_type: pa.DataType):
     elif isinstance(pa_type, pa.FixedSizeListType):
         return Sequence(feature=generate_from_arrow_type(pa_type.value_type), length=pa_type.list_size)
     elif isinstance(pa_type, pa.ListType):
-        return [generate_from_arrow_type(pa_type.value_type)]
+        feature = generate_from_arrow_type(pa_type.value_type)
+        if isinstance(feature, (dict, tuple, list)):
+            return [feature]
+        return Sequence(feature=feature)
```

Comment:
I had to fix that.
There were issues with arrow types like `list of list of struct`.
We have one dataset that has this format: `docred`

Reply:
Great catch!